util/src/test/java/org/apache/jsieve/util/ScriptCheckMailAdapterTest.java

```diff
@@ -0,0 +1,87 @@
+import javax.mail.internet.MimeMessage;
+
+import org.apache.jsieve.ConfigurationManager;
+import org.apache.jsieve.SieveFactory;
+import org.apache.jsieve.mail.ActionRedirect;
+import org.apache.jsieve.parser.generated.Node;
+import org.apache.jsieve.util.check.ScriptCheckMailAdapter;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ScriptCheckMailAdapterTest {
+
+    private static final String FOLDED_SUBJECT_MATCH = "if header :is \"subject\" \"Test with long long long long long long long long long "
+            + "long long long long long long long long long long long very very long folded subject email\" { "
+            + "redirect \"test@test.test\";" + "}";
+    
+    private static final String ENCODED_SUBJECT_MATCH = "if header :is \"subject\" \"Test però £ ù €\" { "
+            + "redirect \"test@test.test\";" + "}";
+
+    private SieveFactory sieveFactory;
+
+
+    @Before
+    public void setUp() throws Exception {
+        this.sieveFactory = new ConfigurationManager().build();
+    }
+
+    @Test
+    public void askedActionMustBeExecutedWhenFoldedHeaderMatchs() throws Exception {
+
+        //Given
+        // Load test message file
+        File file = new File(this.getClass().getResource("FoldedSubjectEmail.eml").getFile());
+        MimeMessage message = new MimeMessage(Session.getDefaultInstance(System.getProperties()),
+            new FileInputStream(file));
+
+        // Associate it to the adapter
+        ScriptCheckMailAdapter testee = new ScriptCheckMailAdapter();
+        testee.setMail(message);
+
+        //When
+        // Parse sieve script
+        Node parsedSieve = sieveFactory.parse(new ByteArrayInputStream(FOLDED_SUBJECT_MATCH.getBytes("UTF-8")));
+
+        // Evaluate the script against the message
+        sieveFactory.evaluate(testee, parsedSieve);
+
+        //Then
+        // Test is OK if we have the "redirect" action
+        assertThat(testee.getActions().get(0)).isInstanceOf(ActionRedirect.class);
+
+    }
+
+    @Test
+    public void askedActionMustBeExecutedWhenEncodedHeaderMatchs() throws Exception {
+
+        //Given
+        // Load test message file
+        File file = new File(this.getClass().getResource("EncodedSubjectEmail.eml").getFile());
+        MimeMessage message = new MimeMessage(Session.getDefaultInstance(System.getProperties()),
+            new FileInputStream(file));
+
+        // Associate it to the adapter
+        ScriptCheckMailAdapter testee = new ScriptCheckMailAdapter();
+        testee.setMail(message);
+
+        //When
+        // Parse sieve script
+        Node parsedSieve = sieveFactory.parse(new ByteArrayInputStream(ENCODED_SUBJECT_MATCH.getBytes("UTF-8")));
+
+        // Evaluate the script against the message
+        sieveFactory.evaluate(testee, parsedSieve);
+
+        //Then
+        // Test is OK if we have the "redirect" action
+        assertThat(testee.getActions().get(0)).isInstanceOf(ActionRedirect.class);
```

Comment:
assertThat(testee.getActions()).hasSize(1);
assertThat(testee.getActions().get(0)).isInstanceOf(ActionRedirect.class);

The test execution result will be more readable in case of empty actions, and will fail if more than one ;-) You will describe more precisely the underlying tested result.